#!/usr/bin/env python

PACKAGE = "ardro_dyn"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameter definition, name, type, level, description, default, min, max
gen.add("new_x",    double_t,   0, "New x:",                    -0.25, -1.5,   1.5)
gen.add("new_y",    double_t,   0, "New y",                     -0.25, -1.5,   1.5)
gen.add("new_z",    double_t,   0, "New z",                     1.0, -1.5,   1.5)
gen.add("vel_x", double_t, 0, "linear velocity X", 0, -3, 3)
gen.add("vel_y", double_t, 0, "linear velocity  Y", 0, -3, 3)
gen.add("vel_z", double_t, 0, "linear velocity  Z", 0, -3, 3)
gen.add("Roll", double_t, 0, "Roll", 0, -3.14, 3.14)
gen.add("Pitch", double_t, 0, "Pitch", 0, -3.14, 3.14)
gen.add("Yaw", double_t, 0, "Yaw", 0, -3.14, 3.14)
gen.add("start",    bool_t,   0, "start",  False)

# Generate the necessary files and exit the program - name of the node, prefix for cpp
# for e.g <name>Config.h, THE LAST PARAMETER SHOULD BE EQUAL TO THE cfg FILE NAME WHITOUT EXTENSION
exit(gen.generate(PACKAGE, "ardro_dyn", "ardroparams"))

